hadoop-ozone/dist/src/main/compose/test-all.sh

```diff
@@ -46,7 +46,8 @@ if [[ "${OZONE_WITH_COVERAGE}" == "true" ]]; then
   cp /tmp/jacoco-combined.exec "$SCRIPT_DIR"/result
 fi
 
-generate_report "acceptance" "${ALL_RESULT_DIR}" "${XUNIT_RESULT_DIR}"
-
+if [[ "${OZONE_ACCEPTANCE_SUITE:-}" != "s3a" ]]; then
+  generate_report "acceptance" "${ALL_RESULT_DIR}" "${XUNIT_RESULT_DIR}"
```

Comment:
Generalized `s3a`-specific condition to check acceptance test type (`robot` or `maven`).  We may be adding more Maven-based tests in the future.

Reply:
Although the [overall job](https://github.com/apache/ozone/actions/runs/8618239388/job/23620931723?pr=6458) passes, there is still a failure because `generate_report` is being called (not sure why this doesn't fail the s3a job):
```
Robot framework is not installed, the reports cannot be generated (sudo pip install robotframework).
...
Error: Process completed with exit code 1.
```

Looks like `OZONE_ACCEPTANCE_TEST_TYPE` is not exported in `acceptance.sh` and not visible in this subshell.

Reply:
Side note, will this also be a problem if devs run `test-all.sh` manually to run all the tests locally?

Reply:
Thanks for pointing out the problem with `OZONE_ACCEPTANCE_TEST_TYPE`.

> will this also be a problem if devs run `test-all.sh` manually to run all the tests locally?

When running `test-all.sh` directly, installing Robot is the developer's responsibility (it was already required before this change).

Reply:
Right, my question is more generally: what is the expected way to run all acceptance tests locally and manually? Ideally running `test-all.sh` (or maybe`acceptance.sh`) should kick off all the acceptance tests locally. After this change, running either of these will only run the robot tests, unless the user knows which variables to export to their shell before running.

Maybe that's ok for now and we can handle that in a follow up task, since there's probably not many people running all the tests this way.

Reply:
> what is the expected way to run all acceptance tests locally and manually?
> ...
> After this change, running either of these will only run the robot tests

Actually, they will also try to run `s3a` tests, but that part will fail, since Hadoop sources will be missing.

We can avoid that by further separating the two kinds of tests.

Reply:
I'm ok to handle this in a follow up task if you want. It works fine in CI and I don't think many people are running the tests this way.

Reply:
Slightly changed `s3a-test.sh` to not fail, rather skip the test if `HADOOP_AWS_DIR` is missing.  This way current behavior is retained when running `acceptance.sh` or `test-all.sh` without any test filters.